src/database/sqlite.rs

```diff
@@ -970,11 +975,49 @@ pub fn migrate(conn: &Connection) -> rusqlite::Result<()> {
     Ok(())
 }
 
+/// A [`DatabaseFactory`] implementation that builds [`SqliteDatabase`].
+///
+/// Each built database is stored in path of format: `<path_root>_<hash>.<path_ext>`
+/// Where `hash` contains identifying data derived with inputs provided by
+/// [`DatabaseFactory::build`] or [`DatabaseFactory::build_with_change`] calls.
+pub struct SqliteDatabaseFactory<P> {
```

Comment:
I had an idea for this, let me know what you think: in theory it should be possible to just have a single database by adding an extra column to all the relevant tables that identifies to which descriptor the rows belong.

So my idea for the factory would be something that wraps the db (probably in an `Arc`) and all the actual `SqliteDatabase` struct would have a member inside to keep track of what id it belongs to. If you create a `SqliteDatabase` directly this could either be `None` or just an empty string.

Then we just have to add an additional `WHERE` clause to the queries and everything should work magically.

Why we may want to do this? Mainly because I guess having a single file makes it easier to work with, backup, etc. And also this opens the possibility to add more methods in the future that iterate over the whole db, which would probably bring a performance benefit to `MultiTracker`.

Disadvantages? I guess the extra `WHERE` clause might cause a small performance hit? But we can also add an index on that column if we feel like it.

Reply:
I think this is a great idea! I would assume that performance wouldn't be too much of an issue since we are already using indexes for almost all tables/queries.

However, I feel like this should be in it's separate issue/PR since we need to change the "nature" of `SqliteDatabase`? What do you think?

Reply:
I think @afilini's suggestion can be done in this PR if you want to give it a shot. Once that's in this should be ready for general review right?

Reply:
@notmandatory Thank you for including this PR in the last call (which I failed to attend :cry: ). I have not touched this PR since the last review because I am skeptical on whether this is actually needed.

My current thinking is that `Database` itself is NOT an ideal model for implementing multi-descriptor, and the correct approach seems to be that of which `bdk_core` is doing. I think more discussion is needed before I would be willing to continue/close this PR.

Reply:
No problem, I think it's fine to leave this one on hold and see how a better approach shapes up with `bdk_core`.